### PR TITLE
VB tuples: support partial names

### PR DESCRIPTION
**Pull request by jcouv**
Vlad's porting of tuples to VB already picked up much code after C# already supported partial names (`CreateTupleTypeSymbol` API, metadata loading), so the remaining code to be updated was limited: binding tuple type and binding tuple expression shouldn't report an error, and some refactoring code had to be updated (in `TypeSyntaxGeneratorVisitor.vb`).

Fixes https://github.com/dotnet/roslyn/issues/13662

@VSadov @dotnet/roslyn-compiler for review

Here are the corresponding changes in C#:
- [Partial names in C#](https://github.com/dotnet/roslyn/pull/12602/files) (ported all the logic and most tests)
- [Extract method refactoring in presence of partial names](https://github.com/dotnet/roslyn/commit/bcc8af40e7616c06c04f2c9ee3c2c30257fe79ef) (ported fix from `CreateTupleTypeSyntax` method)

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb [hunk 1]
```diff
@@ -724,10 +722,6 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Binder.CollectTupleFieldMemberNames(name, i + 1, numElements, elementNames)
                 Next
 
-                If countOfExplicitNames <> 0 AndAlso countOfExplicitNames <> numElements Then
-                    Binder.ReportDiagnostic(diagnostics, syntax, ERRID.ERR_TupleExplicitNamesOnAllMembersOrNone)
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb]: Can we remove this `ERRID` enum value and associated string resource?

File: src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb [hunk 2]
```diff
@@ -350,7 +350,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     AddSpace()
                 End If
 
-                If hasNames Then
+                If hasNames And elementNames(i) IsNot Nothing Then
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb]: `IsNot Nothing` check is already handled below.

**Comment by jcouv** [comment]
There are test failures that I missed and still need to work on.

**Comment by jcouv** [comment]
@dotnet-bot test windows_debug_unit32_prtest  please

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb (hunk 1 above)
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb]: Removed.

**Comment by jcouv** [comment]
I'll need to open a new PR against preview-5 (GitHub's retargeting messed things up).

**Comment by cston** [comment]
LGTM

**Pull request closed without merging**